compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -777,6 +777,14 @@ static Type* getBasicInstantiationType(Type* actualType, Symbol* actualSym,
       return baseType;
   }
 
+  if (formalType->symbol->hasFlag(FLAG_C_PTRCONST_CLASS) &&
+      actualType == dtStringC &&
+      formalType->symbol->hasFlag(FLAG_GENERIC) &&
+      formalSym->hasFlag(FLAG_TYPE_VARIABLE) == false) {
+        return actualType;
+  }
```

Comment:
I don't think you want this code here anymore. It was relevant for earlier approaches but now we are going for `c_string` and `c_ptrConst(c_char)` being different types, right?

Reply:
It was because the signature that accept `c_ptr` and `c_ptrConst` do so in a generic fashion to allow for one signature to work for both `int(8)` and `uint(8)` types.